### DISCUSSION TITLE
Road to OKD 4.8

### DISCUSSION THREAD

**Discussion by vrutkovs** [okd-project/okd#681] [Releases]
OCP code base is about to be frozen, so I'll keep a list of things to fix before we can put OKD 4.8 in stable

TODO:
* [x] Branch installer in [`release-4.8-okd`](https://github.com/vrutkovs/installer/commits/release-4.8-okd)
* [x] Branch machine-config-operator in [`release-4.8-okd`](https://github.com/vrutkovs/machine-config-operator/commits/release-4.8-okd)
* [x] Branch okd-machine-os in [`release-4.8`](https://github.com/openshift/okd-machine-os/commits/release-4.8)
* [x] Create [release-controller configuration](https://github.com/openshift/release/pull/19495)
* [ ] Reconfigure CI to use there branches, create necessary imagestreams
	* [x] [cluster-samples-operator](https://github.com/openshift/release/pull/20055)
	* [ ] [cluster-update-keys](https://github.com/openshift/release/pull/22389)
	* [x] [installer](https://github.com/vrutkovs/installer/commits/release-4.8-okd)
	* [x] [machine-config-operator](https://github.com/vrutkovs/machine-config-operator/commits/release-4.8-okd)
	* [ ] [origin](https://github.com/openshift/release/pull/22393)
	* [x] [ovn-kubernetes](https://github.com/openshift/release/pull/20058)
	* [ ] [operator-marketplace](https://github.com/openshift/release/pull/22390)
* [x] [Update okd-machine-os release-4.8 branch](https://github.com/openshift/okd-machine-os/pull/168) to use stable fedora-coreos-config
* [x] Setup promotion jobs
      okd-machine-os depends on submodule bumps, so no promotion jobs are available anymore
* [x] [Setup OKD 4.7 -> 4.8 periodic upgrade jobs](https://github.com/openshift/release/pull/18877)
* [x] Check [4.7 -> 4.8 upgrade jobs](https://prow.ci.openshift.org/job-history/gs/origin-ci-test/logs/periodic-ci-openshift-release-master-okd-4.8-upgrade-from-4.7-e2e-upgrade-gcp) to evaluate stability
	 * [ ] Fix bugs found in periodics:
     	* [x] [e2e-*-ovn-upgrade are permafailing on service/ingress disruption](https://bugzilla.redhat.com/show_bug.cgi?id=1970985) - 4.9 bug
     	* [x] [release-4.8 cherry-pick of ^](https://github.com/openshift/cluster-network-operator/pull/1167)
     	* [ ] [[sig-network-edge] Application behind service load balancer with PDB is not disrupted](https://bugzilla.redhat.com/show_bug.cgi?id=1929396)
   Worked it around by making an upgrade job without disruption tests
* [x] Fix issues after first release
    * [x] Rebase installer/machine-config-daemon
	* [x] v2s1 images make release unmirrorable
	* [x] [Invalid kubelet used (1.22 instead of 1.21)](https://github.com/openshift/okd/issues/908)
	* [x] Bump image content to make sure latest fixes are included
	* [x] [Update marketplace image](https://github.com/openshift/release/pull/22390)
* [ ] Update README in openshift/okd:
	 * [ ] https://github.com/vrutkovs/installer/blob/fcos-master-devel/docs/user/overview.md#coreos-bootimages - how to find initial VM image
* [ ] Add upgrade edges after fixed 4.8 stable is released

**Comment by alibo** [comment]
@vrutkovs I know you're trying hard to release the new version soon and I really appreciate it. I just want to ask you is there any ETA (even a rough ETA) on when OKD 4.8 will be released?

**Reply by vrutkovs** [reply]
We can release 4.8 when 4.7 -> 4.8 upgrade jobs are setup and reliably pass. The former can be done quickly, the latter depends on found issues. Think couple of weeks for instance?

**Reply by alibo** [reply]
thank you for your reply and clarification 👍

**Comment by alibo** [comment]
I guess the last two items (`Update this repo docs` and `Check 4.7 -> 4.8 upgrade jobs to evaluate stability `) and `cluster-samples-operator` are done? (considering the related PRs are merged)

**Comment by brianotte** [comment]
Is there any new news related to 4.8?

**Reply by m-yosefpor** [reply]
from Vadim's release notes:
> Still no luck with safe 4.7->4.8 upgrade path, blocked by OVN bug #1929396. This prevents us from releasing OKD 4.8 stable

https://bugzilla.redhat.com/show_bug.cgi?id=1929396

**Reply by m-yosefpor** [reply]
see full updates here: https://hackmd.io/YJBn04R5TDi5Sm9XbOGwZA

**Reply by brianotte** [reply]
Hello All,

I did not see any updates in the September 28th meeting on [https://hackmd.io/YJBn04R5TDi5Sm9XbOGwZA](https://hackmd.io/YJBn04R5TDi5Sm9XbOGwZA) related to [OVN bug #1929396](https://bugzilla.redhat.com/show_bug.cgi?id=1929396),  which was blocking 4.8 upgrade from 4.7.

Is there any information that can be shared related to Openshift 4.8? 

Please let us know.

Thank you.

Brian

**Comment by vrutkovs** [comment]
https://github.com/openshift/okd/releases/tag/4.8.0-0.okd-2021-10-01-221835 has been released.

the blocking bug was not resolved, but I found a way to work it around. Note, that router / LB connections may be disrupted (up to 30% of the time in our tests) during update, so plan upgrades accordingly.

**Reply by vrutkovs** [reply]
Next up - [Road to OKD 4.9](https://github.com/openshift/okd/discussions/948)
